translation/message_format.rst

```diff
@@ -0,0 +1,436 @@
+        'organizer_name' => 'Ryan',
+        'organizer_gender' => 'male',
+    ]);
+
+    // prints "John & Jane have invited you for their party!"
+    echo $translator->trans('invition_title', [
+        'organizer_name' => 'John & Jane',
+        'organizer_gender' => 'not_applicable',
+    ]);
+
+The ``{...}`` syntax alternates between "literal" and "code" mode. This allows
+you to use literal text in the select statements:
+
+#. The first ``{organizer_gender, select, ...}`` block starts the "code" mode,
+   which means ``organizer_gender`` is processed as a variable.
+#. The inner ``{... has invited you for her party!}`` block brings you back in
+   "literal" mode, meaning the text is not processed.
+#. Inside this block, ``{organizer_name}`` starts "code" mode again, allowing
+   ``organizer_name`` to be processed as variable.
+
+.. tip::
+
+    While it might seem more logical to only put ``her``, ``his`` or ``their``
+    in the switch statement, it is better to use "complex arguments" at the
+    outermost structure of the message. The strings are in this way better
+    readable for translators and, as you can see in the ``other`` case, other
+    parts of the sentence mgith be influenced by the variable.
+
+Pluralization
+-------------
+
+Another interesting function is ``plural``. It allows you to
+handle pluralization in your messages (e.g. ``There are 3 apples`` vs
+``There is one apple``). The function looks very similair to the ``select`` function:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        nr_of_apples: >
+            {apples, plural,
+                =0    {There are no apples :(}
+                one   {There is one apple...}
+                other {There are # apples!}
+            }
+
+    .. code-block:: xml
+
+        <!-- translations/messages.en.xlf -->
+        <?xml version="1.0"?>
+        <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+            <file source-language="en" datatype="plaintext" original="file.ext">
+                <body>
+                    <trans-unit id="nr_of_apples">
+                        <source>nr_of_apples</source>
+                        <target>{apples, plural, =0 {There are no apples :(} one {There is one apple...} other {There are # apples!}}</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
+
+    .. code-block:: php
+
+        // translations/messages.en.php
+        return [
+            'nr_of_apples' => '{apples, plural,
+                =0    {There are no apples :(}
+                one   {There is one apple...}
+                other {There are # apples!}
+            }',
+        ];
+
+Pluralization rules are actually quite complex and differ for each language.
+For instance, here is the mathematical representation of the Russian
+pluralization rules::
+
+    (($number % 10 == 1) && ($number % 100 != 11))
```

Comment:
TBH this looks very complex and I can't get the value of having this example in the docs and don't understand it anyway 😄 WDYT?

Reply:
After second thought, I  agree. I've now explained it shortly in text. The main reason why I want to include this (and why it was previously included), is that pluralization for english speaking people (and most other countries) seems relatively simple: one (apple) and more than one (apples). So a simple `{number, select, =0 {...}, =1 {...}, other {...}}` seems to be enough. But the reality is that pluralization in many languages is actually very complex and cannot, in any reasonable way, be done using the `select` function.